lib/mongoid/embedded_errors.rb

```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```

Comment:
Can you explain the changes in this file in general? I'm not sure how these are needed for compatibility with newest Mongoid.

Reply:
i dont think we need it as of . there are no strings needed that needs to be frozen

Reply:
I've asked if you could explain changes in the file, all changes, in general. Why are they necessary, as this looks like a lot of unnecessary checks.

I didn't ask about the magic comment - it should definitely be added to all files.

Reply:
Ok, so all of the checks for empty arrays and so on are the changes from latest gemversion on rubygems?

Reply:
yes i havent added any checked